lib/msf/core/exploit/remote/ms_icpr.rb

```diff
@@ -265,6 +279,14 @@ def build_csr(cn:, private_key:, dns: nil, msext_sid: nil, msext_upn: nil, algor
       extensions << OpenSSL::X509::Extension.new(OID_NTDS_CA_SECURITY_EXT, ntds_ca_security_ext.to_der, false)
     end
 
+    unless application_policies.empty?
```

Comment:
I noticed that `application_policies` can be explicitly set to `nil` by the caller and it breaks.
```
msf6 auxiliary(admin/dcerpc/icpr_cert) > run verbose=true CA=my-host-CA RHOSTS=10.1.2.3 username=msfuser password=123456 domain=mydomain.local CERT_TEMPLATE=User on_behalf_of=mydomain\\administrator pfx=/home/n00tmeg/.msf4/loot/20241014152651_default_10.1.2.3_windows.ad.cs_857081.pfx
[*] Running module against 10.1.2.3

[*] 10.1.2.3:445 - Connecting to ICertPassage (ICPR) Remote Protocol
[*] 10.1.2.3:445 - Binding to \cert...
[+] 10.1.2.3:445 - Bound to \cert
[-] 10.1.2.3:445 - Auxiliary failed: NoMethodError undefined method `empty?' for nil:NilClass
[-] 10.1.2.3:445 - Call stack:
[-] 10.1.2.3:445 -   /home/n00tmeg/dev/metasploit-framework/lib/msf/core/exploit/remote/ms_icpr.rb:282:in `build_csr'
[-] 10.1.2.3:445 -   /home/n00tmeg/dev/metasploit-framework/lib/msf/core/exploit/remote/ms_icpr.rb:141:in `do_request_cert'
[-] 10.1.2.3:445 -   /home/n00tmeg/dev/metasploit-framework/lib/msf/core/exploit/remote/ms_icpr.rb:99:in `request_certificate'
[-] 10.1.2.3:445 -   /home/n00tmeg/dev/metasploit-framework/modules/auxiliary/admin/dcerpc/icpr_cert.rb:68:in `block in action_request_cert'
[-] 10.1.2.3:445 -   /home/n00tmeg/dev/metasploit-framework/modules/auxiliary/admin/dcerpc/icpr_cert.rb:83:in `with_ipc_tree'
[-] 10.1.2.3:445 -   /home/n00tmeg/dev/metasploit-framework/modules/auxiliary/admin/dcerpc/icpr_cert.rb:67:in `action_request_cert'
[-] 10.1.2.3:445 -   /home/n00tmeg/dev/metasploit-framework/modules/auxiliary/admin/dcerpc/icpr_cert.rb:53:in `run'
```

Maybe using `blank?` would be safer?
```suggestion
    unless application_policies.blank?
```